apps/heft/src/plugins/DeleteFilesPlugin.ts

```diff
@@ -78,16 +94,26 @@ async function _deleteFilesInnerAsync(pathsToDelete: Set<string>, terminal: ITer
       } catch (error) {
         // If it doesn't exist, we can ignore the error.
         if (!FileSystem.isNotExistError(error)) {
-          // When we encounter an error relating to deleting a directory as if it was a file,
-          // attempt to delete the folder. Windows throws the unlink not permitted error, while
-          // linux throws the EISDIR error.
-          if (FileSystem.isUnlinkNotPermittedError(error) || FileSystem.isDirectoryError(error)) {
-            await FileSystem.deleteFolderAsync(pathToDelete);
-            terminal.writeVerboseLine(`Deleted folder "${pathToDelete}".`);
-            deletedFolders++;
-          } else {
-            throw error;
-          }
+          throw error;
+        }
+      }
+    },
+    { concurrency: Constants.maxParallelism }
+  );
+
+  // Clear out any folders that were encountered during the file deletion process. These
+  // folders should already be empty.
+  await Async.forEachAsync(
```

Comment:
I'm confused what your point here is? These folders should already be empty by virtue of clearing out the children in the code above (for example, in a `**/*` scenario). If a directory is matched by a glob pattern my assumption is that all children are either a) already deleted, or b) will need to be deleted. We also get the advantage of not needing to try-catch the deletion of the target path since we already know what is a directory and what is a file.

Reply:
Yes, though it does accomplish the task it's intended to (it clears out empty folders from patterns like `**/*`, and deletes folders that match the pattern directly). So not sure this is an issue?

Reply:
Figured it might be worth including a code comment since this isn't the traversal order normally associated with deleting a folder tee. Normally when doing a recursive delete folders would be unlinked in post-order tree-walk order to address only being able to delete empty folders.

Reply:
Fair. Will add that.